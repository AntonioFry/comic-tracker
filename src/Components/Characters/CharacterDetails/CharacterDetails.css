.character-details-section {
  background-color: black;
}

.character-details-container {
  display: flex;
  justify-content: left;
  padding: 30px 100px;
  width: 100%;
}

.character-cover-container {
 display: flex;
 justify-content: left;
 width: 100%;
}

.character-thumbnail {
  width: 50%;
}

.character-info-container {
  align-content: flex-start;
  display: flex;
  flex-wrap: wrap;
  width: 100%;
}

.character-info-header {
  color: white;
  width: 100%;
  margin: 0px 0px 20px;
}

.character-info-text {
  color: white;
  width: 100%;
  margin: 0px;
}

.info-and-options {
  align-content: flex-start;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;
  padding: 0px 40px;
}

@media only screen and (max-width: 990px) {
  
  .character-details-container {
    flex-wrap: wrap;
    justify-content: center;
    padding: 30px 200px;
  }

  .info-and-options {
    padding: 0px;
  }

  .save-button {
    width: 80%;
  }

  .character-thumbnail {
    width: 80%;
  }

  .character-info-container {
    align-content: flex-start;
    display: flex;
    text-align: center;
    width: 100%;
    margin: 16px 0px 0px;
  }

}