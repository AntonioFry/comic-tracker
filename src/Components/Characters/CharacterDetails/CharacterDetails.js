import React, { Component } from 'react';

export class CharacterDetails extends Commponent {
  
}