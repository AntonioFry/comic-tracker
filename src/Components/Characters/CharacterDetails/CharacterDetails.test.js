import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import { CharacterDetails } from './CharacterDetails.js';
import { BrowserRouter } from 'react-router-dom';

describe('CharacterDetails', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      addCharcterComics: jest.fn(),
      description: "Bitten by a radioactive spider, high school student Peter Parker gained the speed, strength and powers of a spider. Adopting the name Spider-Man, Peter hoped to start a career using his new abilities. Taught that with great power comes great responsibility, Spidey has vowed to use his powers to help people.",
      id: 1009610,
      name: "Spider-Man",
      removeCharacterComics: jest.fn(),
      removeCharacterId: jest.fn(),
      saveCharacterId: jest.fn(),
      savedCharacters: [],
      setCurrentCharacterComics: jest.fn(),
      thumbnail: { path: "http://i.annihil.us/u/prod/marvel/i/mg/3/50/526548a343e4b", extension: "jpg" }
    }
  })

  it('Should match the snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <CharacterDetails
          thumbnail={{path: "http://i.annihil.us/u/prod/marvel/i/mg/3/50/526548a343e4b", extension: "jpg"}}
          description={"Bitten by a radioactive spider, high school student Peter Parker gained the speed, strength and powers of a spider. Adopting the name Spider-Man, Peter hoped to start a career using his new abilities. Taught that with great power comes great responsibility, Spidey has vowed to use his powers to help people."}
          name={"Spider-Man"}
          savedCharacters={[]}
          id={1009610}
          addCharcterComics={jest.fn()}
          removeCharacterComics={jest.fn()}
          removeCharacterId={jest.fn()}
          saveCharacterId={jest.fn()}
          setCurrentCharacterComics={jest.fn()}
        />
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});