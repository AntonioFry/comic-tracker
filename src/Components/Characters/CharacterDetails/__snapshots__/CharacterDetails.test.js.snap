exports[`CharacterDetails Should match the snapshot 1`] = `
<section
  className="character-details-section"
  style={
    Object {
      "background": "linear-gradient(rgb(0, 0, 0), rgba(0, 0, 0, 0.5)) 0% 0% / cover no-repeat,
            url(http://i.annihil.us/u/prod/marvel/i/mg/3/50/526548a343e4b.jpg)",
      "backgroundSize": "cover",
      "minHeight": "700px",
    }
  }>
  <div
    className="character-details-container">
    <img
      alt="Showing of Spider-Man"
      className="character-thumbnail"
      src="http://i.annihil.us/u/prod/marvel/i/mg/3/50/526548a343e4b.jpg" />
    <div
      className="info-and-options">
      <article
        className="character-info-container">
        <h3
          className="character-info-header">
          Spider-Man
        </h3>
        <h3
          className="character-info-header">
          Description
        </h3>
        <p
          className="character-info-text">
          Bitten by a radioactive spider, high school student Peter Parker gained the speed, strength and powers of a spider. Adopting the name Spider-Man, Peter hoped to start a career using his new abilities. Taught that with great power comes great responsibility, Spidey has vowed to use his powers to help people.
        </p>
      </article>
      <button
        className="disabled-save-button"
        disabled={true}
        onClick={[Function]}>
        SAVE
      </button>
    </div>
  </div>
  <section
    className="comic-rail-container">
    <h2
      className="rail-category-white">
      Spider-Man Comics
    </h2>
    <div
      className="comic-rail" />
  </section>
</section>
`;
